### ISSUE TITLE
Right-hand side must be single expression

### ISSUE THREAD

**Issue by jonsmock** [rbrush/clara-rules#118]
I noticed this rule does not insert the B fact:

```
(defrule calculate-b
  [A]
  =>
  (println "inserting b")
  (clara/insert! (->B)))
```

However, this one does insert it:
```
(defrule calculate-b
  [A]
  =>
  (do (println "inserting b")
      (clara/insert! (->B))))
```

This could very well be intentional; I don't know what the implications are of allowing multiple s-expressions on the right-hand side. From a new-user perspective (as in, I am a brand new user, as of yesterday), I didn't expect this behavior. I wonder if either an exception could be thrown or if multiple s-expressions could be run.

I see there's already a warning about not including a right-hand side [here](https://github.com/rbrush/clara-rules/blob/master/src/main/clojure/clara/rules.clj#L218-L221). This seems like the place to throw for multiple rhs expressions, except that `split-lhs-rhs` only captures the first expression after the separator, [due to destructuring](https://github.com/rbrush/clara-rules/blob/master/src/main/clojure/clara/rules/dsl.clj#L24). Obviously, if you want to run all the s-expressions, you'd just have to destructure with `[lhs [sep & rhs]]` instead.

Anyway, just figured I'd give some feedback. Really liking things so far!

**Comment by mrrodriguez** [comment]
I always thought it'd be more intuitive if Clara would just wrap the RHS in a "do" implicitly. 

I'm fairly sure there is no good reason to force the rule writer to have to wrap all their s expressions in their own "do" block. 

The error is sneaky too. I've been bitten by it before. I'm glad you reported it because I forgot about it. :)

**Comment by rbrush** [comment]
I'm fine with implicitly wrapping the RHS in a do. I personally like the explicitly do block, but that's just a style preference and producing unexpected behavior like it does today is definitely a bug.

I'm going to be traveling over the weekend but can get it in early next week. It would be relatively straightforward to add if someone is interest in taking this on as well.

**Comment by mrrodriguez** [comment]
I guess I was coming from the Drools background where it is just a sequence of statements on the RHS. If there was an explicit parse error thrown for having more than one form on the RHS, I'd still find that reasonable though.

**Comment by rbrush** [comment]
Let's just add an implicit do around the RHS. It seems like the least
surprising behavior and I don't see a good reason not to.

**Comment by jonsmock** [comment]
Thanks for the responses! I can take a shot at this, unless one of you would rather do it. I can probably get to it later this week?

Is [this](https://github.com/rbrush/clara-rules/blob/master/src/test/clojure/clara/test_rules.clj) where you'd want me to add a test for this?

**Comment by rbrush** [comment]
Sounds great, Jonathan! I'd love to pull in a contribution along these lines.

And yes, adding a test to clara.test-rules is the right place. (At some point we might break that test namespace up but that's beyond the scope of this change.)

The simplest change is probably just to have the defrule macro add a (do ....) block to wrap the right-hand side if the right-hand side has multiple elements, and update the split-lhs-rhs as you mentioned. We could take update the engine entirely to pass a sequence of expressions from the right-hand side through  the compilation process, but this would be more invasive and make management of file and line metadata more complicated during the compilation process.

Thanks!

-Ryan

**Comment by mrrodriguez** [comment]
+1 to Jonathan taking a shot at it.  I didn't plan to get a chance to work on it right now either.

**Comment by rbrush** [comment]
Closing since this was merged and marking for the 0.8.9 release
